src/dyn_subgrid/dyncropFileMod.F90

```diff
@@ -165,8 +166,8 @@ subroutine dyncrop_interp(bounds,crop_inst)
     allocate(fertcft_cur(bounds%begg:bounds%endg, cft_lb:cft_ub))
     call fertcft%get_current_data(fertcft_cur)
 
-    if (use_crop) then
-       call collapse_crop_types(wtcft_cur, fertcft_cur, bounds%begg, bounds%endg, verbose = .false.)
+    if (cft_size > 2) then
```

Comment:
Can you explain the need for this change, and the rationale for this particular condition (cft_size > 2)? A comment in the code would help here.

Reply:
With the new "collapse pft" capability, 
`if (use_crop)`
is incorrect here because we intend to read 78-pft data and collapse to 16 pfts when use_crop = .false.

In surfrdMod.F90 we have the following options (including one that calls collapse_crop_types):
`    if ( actual_numcft > 0 ) then
       ! Full crop on file and in CLM
       if ( create_crop_landunit .and. actual_numcft > 2 .and. use_crop)then
          call surfrd_cftformat( ncid, begg, endg, wt_cft, fert_cft, cft_size, natpft_size )  ! Format where CFT's is read in a seperate landunit
       ! Generic crop in file and generic crop in model
       else if ( create_crop_landunit .and. actual_numcft == 2 .and. cft_size == 2 )then
          call surfrd_cftformat( ncid, begg, endg, wt_cft, fert_cft, cft_size, natpft_size )  ! Format where CFT's is read in a seperate landunit
       ! Full crop in file, but model is only with generic crop
       else if ( create_crop_landunit .and. actual_numcft > 2 .and. .not. use_crop)then
          cftsize = actual_numcft
          allocate(array2DCFT (begg:endg,cft_lb:cftsize-1+cft_lb))
          allocate(array2DFERT(begg:endg,cft_lb:cftsize-1+cft_lb))
          call surfrd_cftformat( ncid, begg, endg, array2DCFT, array2DFERT, cftsize, natpft_size )
          call collapse_crop_types(array2DCFT(begg:endg, :), array2DFERT(begg:endg, :), cftsize, begg, endg, &
                                   verbose=.true., sumto=100._r8)
          wt_cft  (begg:,cft_lb:) = array2DCFT (begg:,cft_lb:cft_ub)
          fert_cft(begg:,cft_lb:) = array2DFERT(begg:,cft_lb:cft_ub)
          deallocate(array2DCFT)
          deallocate(array2DFERT)
       else if ( .not. create_crop_landunit )then`

For consistency I will change dyncropFileMod.F90 to the same:
`if (.not. use_crop .and. create_crop_landunit .and. cft_size > 2) then`

I will add a comment in dyncropFileMod.F90.

Long term I see the `cft_size > 2` part becoming obsolete because we will stop using all but the 78-pft datasets.

Reply:
@slevisconsulting just one note on your very last statement "cft_size > 2 part becoming obsolete because we will stop using all but the 78-pft datasets" because of the NWP use-case needs, I'm not sure if this is really the case. For climate we will certainly want to transition to 78-pfts only, but I'm not so sure for NWP cases. I think we will need to support more options for them. Just something we have to start thinking about...

Reply:
@slevisconsulting thank you for adding this comment - this is very helpful. I compared this with the comment at the top of `collapse_crop_types` and the code in that subroutine, and I think the comments and/or logic could still use some reconciliation: 

The comment at the top of `collapse_crop_types` says "Should only be called if using prognostic crops - otherwise, wt_cft is meaningless". The code inside seems to assume that the datasets have 78 pfts. I'm thinking that the comment at the top of that routine should be changed to, "Should only be called if (1) using create_crop_landunit (otherwise, wt_cft is meaningless), and (2) using datasets with the full set of npcropmax crops."

Then the logic that you have here (cft_size > 2) is effectively checking condition (2): I guess we're assuming that datasets always have cft_size <= 2 or cft_size = npcropmax, and never in between. You're not explicitly checking condition (1) (create_crop_landunit) here. I think that's safe at the moment, because we pretty much always use create_crop_landunit, at least by default (is this something a user can change? I'm not sure); the exception is when we're using FATES, but FATES doesn't work with transient right now, so that's currently safe. I'm a little worried about the fragility of this, though: if someone tries to do a transient run with create_crop_landunit false, what would currently happen? Should this be checked somewhere? At the very least, we should probably add to the comment here documenting that assumption (i.e., documenting that, if we get to this point, then we are assuming that create_crop_landunit is true).

Reply:
@slevisconsulting it looks like this issue still needs to be addressed. I think we decided that we should remove the "if ( cft_size > 2 )" around the call to "collapse_crop_types" right? So it looks like that should be done. That possibly means redoing (or rerunning) the non crop tests.

Reply:
Ahh, OK. The diff at the top of this conversation is labeled as "Outdated". I thought I looked at the latest code and saw the same thing, but obviously not. So awesome I'm glad this is done. Thanks for clarifying.